src/charm.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+from charms.observability_libs.v0.kubernetes_service_patch import (  # type: ignore
+    KubernetesServicePatch,
+)
+from charms.prometheus_k8s.v0.prometheus_scrape import (  # type: ignore
+    MetricsEndpointProvider,
+)
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+PROMETHEUS_EDGE_HUB_PORT = 9091
+PROMETHEUS_EDGE_HUB_GRPC_PORT = 9092
+
+
+class PrometheusEdgeHubCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._container_name = self._service_name = "prometheus-edge-hub"
```

Comment:
```suggestion
        name = "prometheus-edge-hub"
        self._service_name = name
        self._container_name = name
```
would make the code more boring (less astonishing). `name` could also be moved up to the outer scope and made a const (ie `DEFAULT_NAME`).

Reply:
Agreed, fixed